crawl-ref/source/cloud.cc

```diff
@@ -1899,3 +1899,24 @@ void end_still_winds()
     env.level_state &= ~LSTATE_STILL_WINDS;
     mpr("The air resumes its normal movements.");
 }
+
+/**
+ * Surround a monster with clouds of a certain type. This also deletes the same
+ * cloud if it's on top of the monster (which will happen as they walk around).
+ */
+void surround_mons_with_cloud(const monster* mons, cloud_type cloud)
```

Comment:
Could you make this work with an `actor *` and refactor `_manage_fire_shield` in `art-func.h` to use it, so that our cloud rings use a unified codepath?